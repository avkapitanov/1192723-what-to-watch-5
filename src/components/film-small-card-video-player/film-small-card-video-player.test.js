import React from "react";
import renderer from "react-test-renderer";
import FilmSmallCardVideoPlayer from "./film-small-card-video-player";

const realUseState = React.useState;
jest
  .spyOn(React, `useState`)
  .mockImplementationOnce(() => realUseState(true))
  .mockImplementationOnce(() => realUseState(false));

const src = `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`;
const poster = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;

describe(`FilmSmallCardVideoPlayer component render correct`, () => {
  it(`With loader`, () => {
    const tree = renderer.create(
        <FilmSmallCardVideoPlayer
          src={src}
          poster={poster}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`With video`, () => {
    const tree = renderer.create(
        <FilmSmallCardVideoPlayer
          src={src}
          poster={poster}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});


