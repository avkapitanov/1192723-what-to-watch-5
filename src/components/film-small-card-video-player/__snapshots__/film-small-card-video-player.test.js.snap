// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmSmallCardVideoPlayer component render correct With loader 1`] = `
Array [
  <svg
    height="57"
    stroke="#fff"
    style={
      Object {
        "left": "50%",
        "marginLeft": "-28px",
        "marginTop": "-60px",
        "position": "absolute",
        "top": "50%",
      }
    }
    viewBox="0 0 57 57"
    width="57"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        strokeWidth="2"
        transform="translate(1 1)"
      >
        <circle
          cx="5"
          cy="50"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="50;5;50;50"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="5;27;49;5"
          />
        </circle>
        <circle
          cx="27"
          cy="5"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="5"
            repeatCount="indefinite"
            to="5"
            values="5;50;50;5"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="27"
            repeatCount="indefinite"
            to="27"
            values="27;49;5;27"
          />
        </circle>
        <circle
          cx="49"
          cy="50"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="50;50;5;50"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="49"
            repeatCount="indefinite"
            to="49"
            values="49;5;27;49"
          />
        </circle>
      </g>
    </g>
  </svg>,
  <video
    autoPlay={true}
    muted={true}
    onCanPlayThrough={[Function]}
    poster="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
    src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
    style={
      Object {
        "display": "none",
      }
    }
  />,
]
`;

exports[`FilmSmallCardVideoPlayer component render correct With video 1`] = `
<video
  autoPlay={true}
  muted={true}
  onCanPlayThrough={[Function]}
  poster="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
  style={
    Object {
      "display": "block",
    }
  }
/>
`;
